packages/thyra/adapters/epetra/test/TestThyraDebugHang.cpp

```diff
@@ -53,45 +61,123 @@ TEUCHOS_UNIT_TEST( ThyraEpetraMultiVector, HangingInParallelDebug )
      os << prefix << "Creating Teuchos::Comm" << endl;
      cerr << os.str ();
    }
-   Teuchos::RCP<const Teuchos::Comm<Teuchos_Ordinal> > comm =
+   RCP<const Teuchos::Comm<Teuchos_Ordinal> > comm =
      Teuchos::DefaultComm<Teuchos_Ordinal>::getComm ();
+   // Make sure that everything is OK on all processes.
+   TEST_ASSERT( ! comm.is_null () );
+   lclSuccess = success ? 1 : 0;
+   reduceAll (*comm, REDUCE_MIN, lclSuccess, outArg (gblSuccess));
+   TEST_EQUALITY( gblSuccess, 1 );
+   if (gblSuccess != 1) {
+     out << "FAILED; some process(es) have a null Teuchos::Comm" << endl;
+     return;
+   }
 
    // Some processors have to have data and some not.
    const int localDim  = myRank % 2;
    const int globalDim = numProcs / 2;
-   Teuchos::RCP<const Epetra_Map> epetra_map;
+   RCP<const Epetra_Map> epetra_map;
    {
      std::ostringstream os;
      os << prefix << "Creating Epetra_Map: localDim=" << localDim << ", globalDim=" << globalDim << endl;
      cerr << os.str ();
    }
-   epetra_map = Teuchos::rcp(new Epetra_Map(globalDim,localDim,0,epetra_comm));
+   epetra_map = rcp (new Epetra_Map (globalDim, localDim, 0, epetra_comm));
+   // Make sure that everything is OK on all processes.
+   TEST_ASSERT( ! epetra_map.is_null () );
+   lclSuccess = success ? 1 : 0;
+   reduceAll (*comm, REDUCE_MIN, lclSuccess, outArg (gblSuccess));
+   TEST_EQUALITY( gblSuccess, 1 );
```

Comment:
This is really just testing Epetra_Map, but that is not a bad test to make sure that Epetra can handle processes with no elements as well.